src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -981,7 +981,9 @@ void primitive_inst::realloc_if_needed(bool prev_execution_skipped) {
                 GPU_DEBUG_TRACE_DETAIL << i << ". " << _impl_params->output_layouts[i].to_string() << std::endl;
                 set_flag(ExecutionFlags::SHAPE_CHANGED);
             } else {
+                bool from_memory_pool = (_outputs[i] && _outputs[i]->from_memory_pool);
                 _outputs[i] = get_network().get_engine().reinterpret_buffer(*_outputs[i], actual_layouts[i]);
+                _outputs[i]->from_memory_pool = from_memory_pool;
```

Comment:
please do this in the reinterpret_buffer

Reply:
Done